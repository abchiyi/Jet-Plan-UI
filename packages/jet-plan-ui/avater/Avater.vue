<template>
    <div class="j-avater" :style="styles" :class="classes">
        <div class="content">
            <slot></slot>
        </div>
    </div>
</template>

<script>
export default {
    name: 'j-avater',
    props: {
        border: {
            type: Boolean,
            default: false,
        },
        circle: {
            type: Boolean,
            default: true,
        },
        size: {
            type: [Number, String],
            default: '2',
        },
    },
    computed: {
        classes() {
            return [this.border ? 'border' : '', this.circle ? 'circle' : ''];
        },
        styles() {
            return {
                '--size': this.size + 'em',
            };
        },
    },
};
</script>

<style >
.j-avater,
.j-avater * {
    transition: 0.5s var(--ease-in-out);
}

.j-avater {
    background: var(--foreground);
    outline: 2px solid var(--border);
    min-height: var(--size);
    min-width: var(--size);
    display: inline-block;
}

.j-avater .content {
    justify-content: center;
    align-items: center;
    height: var(--size);
    display: flex;
}

.j-avater img,
.j-avater {
    border-radius: var(--m-radius);
}

.j-avater.circle img,
.j-avater.circle {
    border-radius: var(--size);
}

.j-avater img {
    height: var(--size);
    width: var(--size);
}
</style>
