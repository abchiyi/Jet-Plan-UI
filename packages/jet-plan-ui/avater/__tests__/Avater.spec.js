import {
    mount
} from '@vue/test-utils'

import Avater from '../Avater.vue'

function cssAttrsStringToObj(str) {
    if (str) {
        const temp = {}
        const attrs = str.split(';');
        for (let n = 0; n < attrs.length; n++) {
            if (attrs[n].trim()) {
                let [key, value] = attrs[n].split(':')
                temp[key.trim()] = value.trim();
            }
        }
        return temp
    } else {
        return str
    }

}

function getStyle(wrapper, attr) {
    return cssAttrsStringToObj(
        wrapper.attributes('style')
    )[attr]
}

describe('Avater', () => {
    it("Document test", async () => {
        const wrapper = mount(Avater);
        expect(wrapper.html()).toMatchSnapshot()

        await wrapper.setProps({
            border: true,
        })
        expect(wrapper.html()).toMatchSnapshot()

        await wrapper.setProps({
            border: true,
            circle: false,
        })
        expect(wrapper.html()).toMatchSnapshot()
        await wrapper.setProps({
            border: true,
            circle: false,
            color: '#61ff73'
        })
        expect(wrapper.html()).toMatchSnapshot()

    })

    it('Class', async () => {
        const wrapper = mount(Avater);

        expect(wrapper.classes()).toContain(
            Avater.name
        )

        expect(wrapper.classes()).toContain(
            'j-avater'
        )
        expect(wrapper.classes()).toContain(
            'circle'
        )
        await wrapper.setProps({
            border: true
        });
        expect(wrapper.classes()).toContain(
            'border'
        )
        await wrapper.setProps({
            border: true,
            circle: false
        });
        expect(wrapper.classes()).not.toContain(
            'circle'
        )
    })

    it('Style value', async () => {
        const wrapper = mount(Avater);

        const defaultSize = {
            size: '2em'
        }

        expect(
            cssAttrsStringToObj(
                wrapper.attributes('style')
            )['--size']
        ).toContain(defaultSize.size)

        await wrapper.setProps({
            size: 4
        })


        expect(
            cssAttrsStringToObj(
                wrapper.attributes('style')
            )['--size']
        ).toContain('4em')

        const color = {
            color: '#61ff73'
        }
        await wrapper.setProps(color)

        expect(
            getStyle(wrapper, '--color')).toContain(color.color)
    })
})
