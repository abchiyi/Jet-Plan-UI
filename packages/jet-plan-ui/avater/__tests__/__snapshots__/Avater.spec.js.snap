// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avater Document test 1`] = `
<div class="j-avater circle" style="--size: 2em; --color: var(--foreground); transition: 1ms ease-in-out;">
  <div class="content"></div>
</div>
`;

exports[`Avater Document test 2`] = `
<div class="j-avater border circle" style="--size: 2em; --color: var(--foreground); transition: 1ms ease-in-out;">
  <div class="content"></div>
</div>
`;

exports[`Avater Document test 3`] = `
<div class="j-avater border" style="--size: 2em; --color: var(--foreground); transition: 1ms ease-in-out;">
  <div class="content"></div>
</div>
`;

exports[`Avater Document test 4`] = `
<div class="j-avater border" style="--size: 2em; --color: #61ff73; transition: 1ms ease-in-out;">
  <div class="content"></div>
</div>
`;
